internal/expr/query.go

```diff
@@ -118,25 +122,25 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 	outputVals := []reflect.Value{}
 	for _, outputArg := range outputArgs {
 		if outputArg == nil {
-			return nil, nil, fmt.Errorf("need map or pointer to struct, got nil")
+			return nil, nil, fmt.Errorf("need map or pointer to struct/primitive type, got nil")
 		}
 		outputVal := reflect.ValueOf(outputArg)
 		k := outputVal.Kind()
 		if k != reflect.Map {
 			if k != reflect.Pointer {
-				return nil, nil, fmt.Errorf("need map or pointer to struct, got %s", k)
+				return nil, nil, fmt.Errorf("need map or pointer to struct/primitive type, got %s", k)
 			}
 			if outputVal.IsNil() {
 				return nil, nil, fmt.Errorf("got nil pointer")
 			}
 			outputVal = outputVal.Elem()
-			k = outputVal.Kind()
-			if k != reflect.Struct && k != reflect.Map {
-				return nil, nil, fmt.Errorf("need map or pointer to struct, got pointer to %s", k)
-			}
```

Comment:
Why are we removing this check? Is it redundant now?

Reply:
This checks that the Kind of the output matches the expected map or struct. This check should now cover all the numerous different kinds that have become valid (there is a separate kind for each size/sign of int).

This error message was actually always a bit redundant, if the user passes a type with an invalid kind it will not be in the query so will be caught there. This message gave a more accurate list of what was valid, though now we would have to change it to "need valid pointer" or something like that since we are not listing all the valid types in error messages elsewhere. The message "need valid pointer" is not really any more useful that "type does not appear in query" so I had decided to remove this check completely.

If you think it would add value for the users I'm happy to add it back in.